server/accounts.go

```diff
@@ -1417,15 +1426,61 @@ func (a *Account) addAllServiceImportSubs() {
 	}
 }
 
-// Helper to determine when to sample.
-func shouldSample(l *serviceLatency) bool {
-	if l == nil || l.sampling <= 0 {
-		return false
+// Helper to determine when to sample. When header has a value, sampling is driven by header
+func shouldSample(l *serviceLatency, c *client) (bool, http.Header) {
+	if l == nil {
+		return false, nil
+	}
+	if l.sampling < 0 {
+		return false, nil
 	}
 	if l.sampling >= 100 {
-		return true
+		return true, nil
+	}
+	if l.sampling > 0 && rand.Int31n(100) <= int32(l.sampling) {
+		return true, nil
+	}
+	h := c.parseState.getHeader()
+	if h == nil {
+		return false, nil
+	}
+	if tId := h.Get("Uber-Trace-Id"); tId != "" {
+		tk := strings.Split(tId, ":")
```

Comment:
Maybe we could put an example as a comment on all the various header types so that makes reading/understanding the code a bit easier?

Reply:
I meant that you put a comment here to show what we are parsing in the lines below, not really have `Get(trcUber)`.. Let me know if it is still not clear.

Reply:
I incorporated comments from RI. Added examples next to the header definition